### PR TITLE
MM-43816_Unable to hit enter to create channel in new Create Channel modal

### PR DESCRIPTION
**Pull request by julmondragon**
#### Summary
New channel modal: enabling enter key to submit channel creation

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43816

#### Screenshots
new channel enter key

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: e2e/cypress/tests/integration/multi_team_and_dm/existing_channel_name_spec.js [hunk 1]
```diff
@@ -62,11 +60,7 @@ function verifyExistingChannelError(newChannelName, makePrivate = false) {
     }
 
     // Type `newChannelName` in the input field for new channel
-    cy.get('#input_new-channel-modal-name').should('be.visible').click().type(newChannelName);
-    cy.wait(TIMEOUTS.HALF_SEC);
-
-    // Click 'Create Channel' button
-    cy.findByText('Create channel').click();
```
Comment (pvev) [inline e2e/cypress/tests/integration/multi_team_and_dm/existing_channel_name_spec.js:69]: shouldn't we keep both scenarios? click create channel button  and enter key press?

**Review by pvev — APPROVED** [review]
LGTM! just one question on the cypress test. :shipit:

File: e2e/cypress/tests/integration/multi_team_and_dm/existing_channel_name_spec.js (hunk 1 above)
Comment (julmondragon) [reply e2e/cypress/tests/integration/multi_team_and_dm/existing_channel_name_spec.js:69]: Sure, there are other tests that will keep using the click event. ie. https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/tests/support/ui/channel.js#L26

**Review by JulienTant — COMMENTED** [review]
One issue with placing this event on the modal rather than on the input is that you can't hit enter in the textarea (for the purpose). Not sure if this is expected?

**Comment by julmondragon** [comment]
> One issue with placing this event on the modal rather than on the input is that you can't hit enter in the textarea (for the purpose). Not sure if this is expected?

Just checked with UX and agreed to disable the submit from the Purpose textarea. Updated!

**Review by JulienTant — APPROVED** [review]
All good @julmondragon thanks!

**Review by stevemudie — APPROVED** [review]
Enter and DONE! Worked as expected @julmondragon!

**Pull request merged**